module-more-apps/src/main/java/com/orange/ods/module/moreapps/data/HttpClient.kt

```diff
@@ -0,0 +1,99 @@
+/*
+ * Software Name: Orange Design System
+ * SPDX-FileCopyrightText: Copyright (c) Orange SA
+ * SPDX-License-Identifier: MIT
+ *
+ * This software is distributed under the MIT license,
+ * the text of which is available at https://opensource.org/license/MIT/
+ * or see the "LICENSE" file for more details.
+ *
+ * Software description: Android library of reusable graphical components
+ */
+
+package com.orange.ods.module.moreapps.data
+
+import android.content.Context
+import com.orange.ods.compose.extension.orElse
+import com.orange.ods.module.moreapps.BuildConfig
+import okhttp3.Cache
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.Response
+import okhttp3.logging.HttpLoggingInterceptor
+import timber.log.Timber
+
+internal class HttpClient(context: Context, private val networkManager: NetworkManager) {
+
+    companion object {
+        private const val CACHE_SIZE: Long = 5 * 1024 * 1024 // 5 MB
+    }
+
+    private val cache = Cache(context.cacheDir, CACHE_SIZE)
+
+    fun build() = OkHttpClient.Builder()
+        .addInterceptor(
+            HttpLoggingInterceptor()
+                .apply {
+                    if (BuildConfig.DEBUG) {
+                        setLevel(HttpLoggingInterceptor.Level.BODY)
+                    }
+                },
+        )
+        .addInterceptor(CacheLoggingInterceptor())
+        .addInterceptor(OfflineInterceptor())
+        .addNetworkInterceptor(OnlineInterceptor())
+        .cache(cache)
+        .build()
+
+    /**
+     * Read from cache for 60 seconds even if there is Internet connection.
+     * Remove "no-cache" attribute from "Pragma" and "Cache-Control" response headers
+     * otherwise we won't be able to fetch response from cache when network is offline.
+     */
+    private inner class OnlineInterceptor : Interceptor {
+        override fun intercept(chain: Interceptor.Chain): Response {
+            val response = chain.proceed(chain.request())
+            val maxAge = 60 // 60 sec
+            return response.newBuilder()
+                .header("Cache-control", "public, max-age=$maxAge")
```

Comment:
```suggestion
                .header("Cache-Control", "public, max-age=$maxAge")
```

It's only cosmetic because the check is case insensitive.